### PR TITLE
Simplex/pyramid/wedge: add DG test

### PR DESCRIPTION
**Pull request by peterrum**
~Depends on #11401 and #11419.~

### CODE REVIEW

File: tests/simplex/step-12c.cc [hunk 1]
```diff
@@ -0,0 +1,1158 @@
...
+          };
+
+          DGMethod<3> dgmethod(scratch_data);
+          dgmethod.run();
+        }
+
+      // pyramid
+      for (unsigned int i = 1; i <= 1; ++i)
+        {
+          continue;
```
Comment (peterrum) [inline tests/simplex/step-12c.cc:1061]: I can enable the pyramid and the wedge test once #11401 is merged.

Comment (peterrum) [reply tests/simplex/step-12c.cc]: Matrix-free and matrix-based results for linear and quadratic discontinuous simplex. hypercube, pyramid, and wedge elements coincide.

**Comment by peterrum** [comment]
@kronbichler Ready once #11400, #11401, and #11848 are merged.

**Comment by peterrum** [comment]
Ready from my side. Once, this is merge we have tests showing that `MatrixFree` is working for DG and mixed meshes!

File: source/fe/mapping_fe.cc [hunk 2]
```diff
@@ -1717,7 +1716,9 @@ namespace internal
                   typename FEValuesBase<dim>::ExcAccessToUninitializedField(
                     "update_contravariant_transformation"));
 
-                for (unsigned int i = 0; i < output.size(); ++i)
+                for (unsigned int i = 0;
+                     i < std::min(input.size(), output.size());
+                     ++i)
```
Comment (kronbichler) [inline source/fe/mapping_fe.cc:1721]: Can you add a short comment that explains the case when these two are different (wedge/pyramid case)?

**Review by kronbichler — APPROVED** [review]
Excellent! This is great news.

File: source/fe/mapping_fe.cc (hunk 2 above)
Comment (peterrum) [reply source/fe/mapping_fe.cc]: I think now it should be better.

**Review by kronbichler — APPROVED** [review]
Much better indeed!

**Pull request merged**